packages/core/test/insured-bridge/AVM_BridgeDepositBox.js

```diff
@@ -0,0 +1,228 @@
+// This set of tests validates the BridgeDepositBox AVM (Abribtum) specific logic such as L1->L2 function calls and the
+// use of the AVM token bridge. Bridge Deposit logic is not directly tested. For this see the BridgeDepositBox.js tests.
+
+const hre = require("hardhat");
+const { web3 } = require("hardhat");
+const { toWei } = web3.utils;
+const { getContract, assertEventEmitted } = hre;
+
+const { didContractThrow } = require("@uma/common");
+
+const { applyL1ToL2Alias } = require("./helpers/ArbitrumHelper");
+
+const { assert } = require("chai");
+
+// Tested contract
+const BridgeDepositBox = getContract("AVM_BridgeDepositBox");
+const { deployContractMock } = require("./helpers/SmockitHelper");
+
+// Fetch the artifacts to create a mock arbitrum gateway router
+const { L2GatewayRouter__factory } = require("arb-ts");
+
+// Helper contracts
+const Token = getContract("ExpandedERC20");
+const Timer = getContract("Legacy_Timer");
+
+// Contract objects
+let depositBox, l1TokenAddress, l2Token, timer, l2GatewayRouterMock;
+
+// As these tests are in the context of l2, we dont have the deployed notion of an "L1 Token". The L1 token is within
+// another domain (L1). To represent this, we can generate a random address to represent the L1 token.
+l1TokenAddress = web3.utils.toChecksumAddress(web3.utils.randomHex(20));
+
+const minimumBridgingDelay = 60; // L2->L1 token bridging must wait at least this time.
+const depositAmount = toWei("50");
+const slowRelayFeePct = toWei("0.005");
+const instantRelayFeePct = toWei("0.005");
+const quoteTimestampOffset = 60; // 60 seconds into the past.
+
+describe("AVM_BridgeDepositBox", () => {
+  // Account objects
+  let accounts, deployer, user1, bridgeAdminCrossDomainAlias, bridgeAdmin, rando, bridgePool;
+
+  before(async function () {
+    accounts = await web3.eth.getAccounts();
+    [deployer, user1, bridgeAdmin, rando, bridgePool] = accounts;
+
+    timer = await Timer.new().send({ from: deployer });
+
+    // Generate the aliased address of bridgeAdmin. This is the address that'll get sent via the canonical bridge when
+    // the bridge admin calls methods to L2. See the wallet with some eth. Enable account impersonation in hre.
+    bridgeAdminCrossDomainAlias = applyL1ToL2Alias(bridgeAdmin);
+    await web3.eth.sendTransaction({ from: deployer, to: bridgeAdminCrossDomainAlias, value: toWei("1") });
+    await hre.network.provider.request({
```

Comment:
OOC: Why do you need to impersonate this account? Can't you just send this address some ETH and then send txns from it?

Reply:
I thought you could! spent a good few hours trying to figure out why I cant just define the `from` and then send but it actually _requires_ this kind of impersonation to send in this context for some reason. I think the way we've done impersonation before was: 1) for view methods mainly and 2) used smockit such that a fair amount of this was abstracted away.